### PR TITLE
feat(adapters): added extra validation in adapters

### PR DESCRIPTION
**Pull request by sereneATL**
#### What this PR does / why we need it:
- added extra validation and error handling in adapters
- standardise error handling
#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: adapters/src/iexcloudforex.ts [hunk 1]
```diff
@@ -48,6 +48,12 @@ export default async function (symbols: string[], apiToken: string): Promise<Ass
 
     return newAssetPrice(asset, price, 'USD', x.timestamp)
   })
+
+  if (allAssetPrices.length !== symbols.length) {
+    throw Error('iexcloudforex.missingTickerSymbol ')
```
Comment (fuxingloh) [inline adapters/src/iexcloudforex.ts]:
```suggestion
    throw Error('iexcloudforex.missingTickerSymbol')
```

File: adapters/src/nasdaq.ts [hunk 2]
```diff
@@ -52,6 +52,10 @@ async function fetchAsset (symbol: string, authToken: string): Promise<AssetPric
   const priceObject = Array.isArray(response.data) ? response.data[0] : response.data
   const timestamp = new BigNumber(processTimestamp(priceObject.timestamp))
 
+  if (symbol !== priceObject.symbol) {
+    throw new Error('nasdaq.mismatchedTickerSymbol ')
```
Comment (fuxingloh) [inline adapters/src/nasdaq.ts]:
```suggestion
    throw new Error('nasdaq.mismatchedTickerSymbol')
```

File: adapters/src/dex-coingecko.ts [hunk 3]
```diff
@@ -17,7 +17,11 @@ export async function getBitcoinPrice (): Promise<BigNumber> {
   )
 
   if (res.status !== 200) {
-    throw new Error('dex-cmc.invalidCoinGeckoResponse ')
+    throw new Error('dex-coingecko.invalidCoinGeckoResponse')
+  }
+
+  if (!(Object.keys(res.data).includes('bitcoin'))) {
```
Comment (fuxingloh) [inline adapters/src/dex-coingecko.ts]:
```suggestion
  if (res.data['bitcoin'] === undefined)) {
```

File: adapters/src/dex-cmc.ts [hunk 4]
```diff
@@ -21,7 +21,11 @@ export async function getBitcoinPrice (apiToken?: string): Promise<BigNumber> {
   )
 
   if (res.status !== 200) {
-    throw new Error('dex-cmc.invalidCMCResponse ')
+    throw new Error('dex-cmc.invalidCMCResponse')
+  }
+
+  if (!(Object.keys(res.data.data).includes('BTC'))) {
```
Comment (fuxingloh) [inline adapters/src/dex-cmc.ts]:
```suggestion
  if (res.data.data['BTC'] === undefined) {
```

File: adapters/src/coingecko.ts [hunk 5]
```diff
@@ -16,7 +16,15 @@ export default async function (symbols: string[]): Promise<AssetPrice[]> {
   const ids = symbols.map(symbol => MAPPING[symbol])
   const res = await fetchAsJson(`${URL}?ids=${ids.join(',')}&vs_currencies=usd`)
 
+  if (Object.keys(res.data).length !== symbols.length) {
+    throw new Error('coingecko.missingTickerSymbol')
+  }
+
   return symbols.map((symbol: string): AssetPrice => {
+    if (!(Object.keys(res.data).includes(MAPPING[symbol]))) {
```
Comment (fuxingloh) [inline adapters/src/coingecko.ts]: Same for this

**Review by fuxingloh — DISMISSED** [review]
Minor change suggestion

File: adapters/src/coingecko.ts (hunk 5 above)
Comment (sereneATL) [reply adapters/src/coingecko.ts]: thanksss, i've added all the changes in the next commit 🙏

**Review by joeldavidw — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**